### PR TITLE
[UI] Redact envs for cluster YAMLs

### PR DESCRIPTION
**Pull request by Michaelvll**
The cluster YAMLs in the UI have `envs` in it, which may leak user's own secrets. For security reason, we now redact the values for envs for now as a first step.

Tested (run the relevant ones):

- [ ] Code formatting: install pre-commit (auto-check on commit) or `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `/smoke-test` (CI) or `pytest tests/test_smoke.py` (local)
- [ ] Relevant individual tests: `/smoke-test -k test_name` (CI) or `pytest tests/test_smoke.py::test_name` (local)
- [ ] Backward compatibility: `/quicktest-core` (CI) or `pytest tests/smoke_tests/test_backward_compat.py` (local)

### CODE REVIEW

File: sky/task.py [hunk 1]
```diff
@@ -1314,8 +1453,20 @@ def add_if_not_none(key, value, no_empty: bool = False):
...
         add_if_not_none('envs', self.envs, no_empty=True)
 
+        # Add secrets with redaction if requested
+        secrets = self.secrets
+        if secrets:
+            if redact_secrets:
+                secrets = {
+                    k: '<redacted>' if isinstance(v, str) else v
+                    for k, v in secrets.items()
+                }
```
Comment (aylei) [inline sky/task.py]:
```suggestion
        if secrets and redact_secrets:
            secrets = {
                k: '<redacted>' if isinstance(v, str) else v
                for k, v in secrets.items()
            }
```

Simplified a bit?

File: sky/cli.py [hunk 2]
```diff
@@ -231,6 +231,22 @@ def _parse_env_var(env_var: str) -> Tuple[str, str]:
     return ret[0], ret[1]
 
 
+def _parse_secret_var(secret_var: str) -> Tuple[str, str]:
+    """Parse secret vars into a (KEY, VAL) pair."""
+    if '=' not in secret_var:
```
Comment (aylei) [inline sky/cli.py:236]: I'm wondering a corner case where the secret var contains `=` (e.g. a based encoded token) and that might lead to unexpected behavior. We need a unit test for this

**Review by aylei — APPROVED** [review]
Did not look into the js code yet, the python code LGTM. Thanks @Michaelvll !

File: sky/cli.py (hunk 2 above)
Comment (aylei) [reply sky/cli.py:236]: Oh nvm, it should be fine since it will only appears in values, which must come after the `=` of `KEY=VAL`
